src/hooks/useImagePreSetup.ts

```diff
@@ -0,0 +1,144 @@
+import { useEffect, useState } from 'react';
+import { fileImgToWebP } from '../utils';
+
+interface UseImagePreSetupProps {
+  imageFiles: File[] | null;
+  convertToWebP?: boolean;
+  webPQuality?: number;
+}
+
+interface UseImagePreSetupReturns {
+  previewUrls: (string | null)[];
+  isLoading: boolean;
+  isError: boolean;
+  webpImages: (Blob | null)[];
+  originalImages: File[] | null;
+}
+
+interface ProcessedFileResult {
+  webpBlob: Blob | null;
+  previewUrl: string | null;
+}
+
+const convertFile = async (
+  file: File,
+  convertHandler: (file: File) => Promise<ProcessedFileResult>,
+  idx: number
+): Promise<ProcessedFileResult> => {
+  try {
+    return await convertHandler(file);
+  } catch (error) {
+    console.error(`${idx + 1}번째 파일 처리 중 오류 발생:`, error);
+    return { webpBlob: null, previewUrl: null };
+  }
+};
+const DEFAULT_WEBP_QUALITY = 0.8;
+
+const validateWebPQuality = (convertToWebP: boolean, webPQuality: number) => {
+  if (!convertToWebP && webPQuality !== DEFAULT_WEBP_QUALITY) {
+    console.warn(
+      'webPQuality`는 WebP로의 변환 품질을 설정하는 옵션입니다. `convertToWebP`를 true로 설정해야만 `webPQuality`가 적용됩니다.'
+    );
+  }
+
+  if (webPQuality < 0 || webPQuality > 1) {
+    console.warn(
+      `webPQuality 값이 유효 범위(0 ~ 1)를 벗어나 기본값(${DEFAULT_WEBP_QUALITY})이 사용됩니다.`
+    );
+    webPQuality = DEFAULT_WEBP_QUALITY;
+  }
+
+  return webPQuality;
+};
```

Comment:
```suggestion
const validateWebPQuality = (convertToWebP: boolean, webPQuality: number) => {
  if (!convertToWebP) {
    if(webPQuality !== DEFAULT_WEBP_QUALITY){
      console.warn(
        'webPQuality`는 WebP로의 변환 품질을 설정하는 옵션입니다. `convertToWebP`를 true로 설정해야만 `webPQuality`가 적용됩니다.'
      );
      }
      // convertToWebP가 false인 경우에는 의미가 없으므로 기본값으로 리턴
    return DEFAULT_WEBP_QUALITY;
  }

  if (webPQuality < 0 || webPQuality > 1) {
    console.warn(
      `webPQuality 값이 유효 범위(0 ~ 1)를 벗어나 기본값(${DEFAULT_WEBP_QUALITY})이 사용됩니다.`
    );
    return DEFAULT_WEBP_QUALITY;
  }

  return webPQuality;
};
```

early return 해주는건 어때유?

Reply:
[commit] [8672915](https://github.com/frontend-opensource-project/use-react-hooks/pull/58/commits/86729152edfd381a551456b3156779df380537ab)
@Choozii  반영했습니다!